Sources/LogInOut.php

```diff
@@ -97,13 +97,9 @@ function Login2()
 
 	if (isset($_GET['sa']) && $_GET['sa'] == 'salt' && !$user_info['is_guest'])
 	{
-		if (isset($_COOKIE[$cookiename]) && preg_match('~^a:[34]:\{i:0;i:\d{1,7};i:1;s:(0|128):"([a-fA-F0-9]{128})?";i:2;[id]:\d{1,14};(i:3;i:\d;)?\}$~', $_COOKIE[$cookiename]) === 1)
+		if (isset($_COOKIE[$cookiename]) && preg_match('~^\[\d{1,7},"([a-fA-F0-9]{128})?",\d{1,14}(,[0-3])?\]$~', $_COOKIE[$cookiename]) === 1)
```

Comment:
both pattern check, the old and new one, are wrong.
Since my update the cookie look like so:
`{"0":1,"1":"7a30e3ff27d981d93f57fb0aaa2065b96c3144535fbbeebb3aa1e9fa7e68d771e127f8283fb66ca828e8cdf543ff9b14eccd5a5b731c8c885f683a266ce4e217","2":1697263514,"3":3,"path":"\/SMF2.1\/"}`

Reply:
albertlast - I didn't pick up on the fact that you redefined the data portion of the cookie.  I think you should revisit that decision.